airflow/configuration.py

```diff
@@ -229,6 +234,14 @@ def validate(self):
 
         self.is_validated = True
 
+    def _validate_enums(self):
+        """Validate that enum type config has an accepted value"""
+        for setting, enum_class in self.enums.items():
+            if self.has_option('core', setting):
+                value = self.get("core", setting)
+                if not enum_class.is_valid(value):
```

Comment:
For an actaul enum (i.e. one inheriting from `enum.Enum`) the test for this is

```
try:
    enum_class(value)
except ValueError:
    raise AirflowConfigException(f"{value} is not an accepted config for {setting}") from None
```

Reply:
yeah that should be done after https://github.com/apache/airflow/pull/5302